airflow/www/extensions/init_jinja_globals.py

```diff
@@ -69,10 +70,13 @@ def prepare_jinja_globals():
             "git_version": git_version,
             "k8s_or_k8scelery_executor": IS_K8S_OR_K8SCELERY_EXECUTOR,
             "rest_api_enabled": False,
-            "auth_manager": get_auth_manager(),
             "config_test_connection": conf.get("core", "test_connection", fallback="Disabled"),
         }
 
+        # Extra global specific to auth manager
+        extra_globals["auth_manager"] = get_auth_manager()
+        extra_globals["DagDetails"] = DagDetails
```

Comment:
But looking at the code, `dag.html` is used a lot in multiple views. So I would have to compute and pass the value `can_edit` in multiple views. I am not sure this is better either. WDYT?

Reply:
It indeed sounds strange to pass authmanager as JINJA global context (especially that it is only used for checking "can_edit".

I think it could be easily fixed for all those multiple views by injecting "can_edit" in `baseviews.py` based on whether `dag` is already present in the arguments (all the views that extend dag_html must have `dag` present). 

Smth like:

````python
    def render_template(self, template, **kwargs):
        """
        Use this method on your own endpoints, will pass the extra_args
        to the templates.

        :param template: The template relative path
        :param kwargs: arguments to be passed to the template
        """
        kwargs["base_template"] = self.appbuilder.base_template
        kwargs["appbuilder"] = self.appbuilder 

        # THIS
        if "dag" in kwargs:  
           kwargs["can_edit"] = get_auth_manager().is_authorized_dag(method="PUT", details=DagDetails(id=kwargs["dag"].dag_id))
        
        return render_template(
            template, **dict(list(kwargs.items()) + list(self.extra_args.items()))
        )

```
Or it could be added in all the views individualy in extra_args

Reply:
Nice, I like it. I no longer pass the auth manager to views in `prepare_jinja_globals` but in `render_template ` now. For context, we still need to pass it to the view because we use it, for instance, to figure whether the user is logged in

Reply:
From the tests it does not seem to work:

```
FAILED tests/www/views/test_views_acl.py::test_success[rendered-templates-all-dag-user] - jinja2.exceptions.UndefinedError: 'can_edit' is undefined
FAILED tests/www/views/test_views_acl.py::test_success[task] - AssertionError: Couldn't find 'Task Instance Details'
FAILED tests/www/views/test_views_acl.py::test_success[xcom] - jinja2.exceptions.UndefinedError: 'can_edit' is undefined
FAILED tests/www/views/test_views_acl.py::test_success[grid-data-for-readonly-role] - AssertionError: Couldn't find 'runme_1'
```

Reply:
It seems `dag.html` is also used to render individual DAG in a view where there are multiple DAGs. Therefore, there is no `kwargs['dag']` but `kwargs['dags']`. I reverted the changes unless there is a better solution

Reply:
Never mind, I confused myself. I figured it out :)